cmd/querier/main.go

```diff
@@ -75,18 +75,24 @@ func main() {
 	}
 	defer chunkStore.Stop()
 
-	queryable := querier.NewQueryable(dist, chunkStore)
-	engine := promql.NewEngine(queryable, nil)
-	api := v1.NewAPI(engine, querier.DummyStorage{Queryable: queryable},
-		querier.DummyTargetRetriever{}, querier.DummyAlertmanagerRetriever{})
-	promRouter := route.New(func(r *http.Request) (context.Context, error) {
-		return r.Context(), nil
-	}).WithPrefix("/api/prom/api/v1")
+	sampleQueryable := querier.NewQueryable(dist, chunkStore, false)
+	metadataQueryable := querier.NewQueryable(dist, chunkStore, true)
+
+	engine := promql.NewEngine(sampleQueryable, nil)
+	api := v1.NewAPI(
+		engine,
+		metadataQueryable,
+		querier.DummyTargetRetriever{},
+		querier.DummyAlertmanagerRetriever{},
```

Comment:
What does it mean to give the API dummy retrievers?

Reply:
The normal Prometheus API can provide a list of targets (e.g. http://demo.robustperception.io:9090/api/v1/targets) and configured Alertmanagers (http://demo.robustperception.io:9090/api/v1/alertmanagers). We don't need/have this information for Cortex, so we pass in dummy implementations that just give empty lists.